hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/OnHeapMapScanP.java

```diff
@@ -0,0 +1,242 @@
+                Object[] row = prepareRow(
+                        iteratorExec.getKey(),
+                        iteratorExec.getKeyData(),
+                        iteratorExec.getValue(),
+                        iteratorExec.getValueData()
+                );
+                done = iteratorExec.done();
+                return row;
+            }
+            return null;
+        }
+
+        public void init(SimpleExpressionEvalContext ctx) {
+            this.row = MapScanRow.create(
+                    node.getKeyDescriptor(),
+                    node.getValueDescriptor(),
+                    node.getFieldPaths(),
+                    node.getFieldTypes(),
+                    mapContainer.getExtractors(),
+                    ctx.getSerializationService()
+            );
+            this.ctx = ctx;
+            this.iteratorExec = new MapScanExecIterator(mapContainer, localPartitionsIterator, ctx.getSerializationService());
+        }
+
+
+        /**
+         * Prepare the row for the given key and value:
+         * 1) Check filter
+         * 2) Extract projections
+         *
+         * @param rawKey       key as object, might be null
+         * @param rawKeyData   key as data, might be null
+         * @param rawValue     value as object, might be null
+         * @param rawValueData value as data, might be null
+         * @return Row that is ready for processing by parent operators or {@code null} if the row hasn't passed the filter.
+         */
+        protected Object[] prepareRow(Object rawKey, Data rawKeyData, Object rawValue, Data rawValueData) {
+            row.setKeyValue(rawKey, rawKeyData, rawValue, rawValueData);
+
+            // Filter.
+            if (filter != null && TernaryLogic.isNotTrue(filter.evalTop(row, ctx))) {
+                return null;
+            }
+
+            if (projections.size() == 0) {
+                return new Object[0];
+            }
+
+            HeapRow heapRow = new HeapRow(projections.size());
+
+            for (int j = 0; j < projections.size(); j++) {
+                Object projectRes = evaluate(projections.get(j), row, ctx);
+                heapRow.set(j, projectRes);
+            }
+
+            return heapRow.getValues();
+        }
+    }
+
+    public static ProcessorMetaSupplier onHeapMapScanP(@Nonnull JetMapScanMetadata mapScanMetadata) {
+        return new OnHeapMapScanMetaSupplier(mapScanMetadata);
+    }
+
+    public static class OnHeapMapScanMetaSupplier implements ProcessorMetaSupplier {
+        private final JetMapScanMetadata mapScanMetadata;
+        private transient Map<Address, List<Integer>> addrToPartitions;
+
+        public OnHeapMapScanMetaSupplier(@Nonnull JetMapScanMetadata mapScanMetadata) {
+            this.mapScanMetadata = mapScanMetadata;
+        }
+
+        @Override
+        public void init(@Nonnull Context context) throws Exception {
+            Set<Partition> partitions = context.jetInstance()
```

Comment:
I would rather create the assignments based on actual addresses (in `get()`) - maybe logic from `JoinByEquiJoinProcessorSupplier` can be reused or shared.